File: app/src/lib/money.ts
/** Tüm tutarlar DB'de INTEGER kuruş. UI'da ₺ olarak gösterilir. */

const fmt = new Intl.NumberFormat("tr-TR", {
  style: "currency",
  currency: "TRY",
  maximumFractionDigits: 0,
});

const fmtPrecise = new Intl.NumberFormat("tr-TR", {
  style: "currency",
  currency: "TRY",
  minimumFractionDigits: 2,
});

export function formatKurus(kurus: number | null | undefined): string {
  if (kurus == null) return "—";
  const lira = kurus / 100;
  return Number.isInteger(lira) ? fmt.format(lira) : fmtPrecise.format(lira);
}

/** Gizlilik Modu açıkken parasal değeri "********" ile maskeler. */
export function formatKurusPrivate(
  kurus: number | null | undefined,
  hidden: boolean
): string {
  return hidden ? "********" : formatKurus(kurus);
}

/** "14.500" | "14500" | "14500,50" → kuruş. Geçersizse null. */
export function parseLiraInput(raw: string): number | null {
  const cleaned = raw.trim().replace(/[₺\s]/g, "").replace(/\./g, "").replace(",", ".");
  if (!cleaned) return null;
  const n = Number(cleaned);
  if (!Number.isFinite(n) || n < 0) return null;
  return Math.round(n * 100);
}

File: mobile/lib/core/utils/money.dart
import 'package:intl/intl.dart';

/// Tüm tutarlar DB'de INTEGER kuruş (masaüstüyle aynı sözleşme). UI'da ₺ olarak gösterilir.
final NumberFormat _fmt = NumberFormat.currency(locale: 'tr_TR', symbol: '₺', decimalDigits: 0);
final NumberFormat _fmtPrecise = NumberFormat.currency(locale: 'tr_TR', symbol: '₺', decimalDigits: 2);

String formatKurus(int? kurus) {
  if (kurus == null) return '—';
  final lira = kurus / 100;
  return lira == lira.roundToDouble() ? _fmt.format(lira) : _fmtPrecise.format(lira);
}

/// Gizlilik Modu açıkken parasal değeri maskeler (masaüstü formatKurusPrivate ile aynı).
String formatKurusPrivate(int? kurus, bool hidden) => hidden ? '********' : formatKurus(kurus);

/// "14.500" | "14500" | "14500,50" → kuruş. Geçersizse null.
int? parseLiraInput(String raw) {
  final cleaned = raw.trim().replaceAll(RegExp(r'[₺\s]'), '').replaceAll('.', '').replaceAll(',', '.');
  if (cleaned.isEmpty) return null;
  final n = double.tryParse(cleaned);
  if (n == null || n < 0) return null;
  return (n * 100).round();
}
